### DISCUSSION TITLE
v5 How to migration:up change column name ? with postgresql

### DISCUSSION THREAD

**Discussion by productdevbook** [mikro-orm/mikro-orm#3158] [Q&A]
Problem
```
  jse_cause: InvalidFieldNameException: alter table "user" rename column "lastlogintime" to "lastLoginAt"; - column "lastlogintime" does not exis
  ```
  
  
  ```ts
  import { Migration } from '@mikro-orm/migrations';

export class Migration20220527162816 extends Migration {

  async up(): Promise<void> {
    this.addSql('alter table "user" add column "lastLoginAt" bigint null default null;');
    this.addSql('comment on column "user"."lastLoginAt" is \'last login time\';');
    this.addSql('alter table "user" drop column "lastLoginTime";');
    this.addSql('alter table "user" rename column "lastlogintime" to "lastLoginAt";');

    this.addSql('alter table "userDevice" add column "lastLoginAt" bigint null;');
    this.addSql('comment on column "userDevice"."lastLoginAt" is \'last login time\';');
    this.addSql('alter table "userDevice" drop column "lastLoginTime";');
    this.addSql('alter table "userDevice" rename column "lastlogintime" to "lastLoginAt";');
  }

  async down(): Promise<void> {
    this.addSql('alter table "user" add column "lastLoginTime" bigint null default null;');
    this.addSql('comment on column "user"."lastLoginTime" is \'last login time\';');
    this.addSql('alter table "user" drop column "lastLoginAt";');
    this.addSql('alter table "user" rename column "lastloginat" to "lastLoginTime";');

    this.addSql('alter table "userDevice" add column "lastLoginTime" bigint null;');
    this.addSql('comment on column "userDevice"."lastLoginTime" is \'last login time\';');
    this.addSql('alter table "userDevice" drop column "lastLoginAt";');
    this.addSql('alter table "userDevice" rename column "lastloginat" to "lastLoginTime";');
  }

}

```

**Comment by productdevbook** [comment]
Is there a solution for this problem, it is tiring us a lot. I have to reset the database every time

**Reply by Langstra** [reply]
It looks like that column does not exist any more at that point. You also seem to drop that column in the third query? Execute the queries locally one by one to see what happens exactly.

**Reply by productdevbook** [reply]
We are loading the new database. Then we change a table name and this error pops up.

**Reply by B4nan** [reply]
I told you several times already, if you want help, you need to start providing complete reproductions. We can't help you with this, you presented just the generated output from schema comparator, we need to see what caused it - the entity state before the change and after, and ideally the database state as well, I kinda expect you have the database a bit handcrafted, otherwise you would probably not have those issues.

The issue is quite apparent, the `lastlogintime` column is there both lowercase as well as pascal case, those are different columns, one gets renamed, one gets dropped and recreated. The question is how did that happen, and without the full entity definition before and after, and ideally the database snapshot, we can't really do anything.
